src/index.ts

```diff
@@ -1,3 +1,73 @@
-export const add = (a: number, b: number) => {
-  return a + b
+import { ApiPromise, WsProvider } from '@polkadot/api'
+import { options } from './options'
+
+/**
+ * The Mangata class defines the `getInstance` method that lets clients access the unique singleton instance. Design pattern Singleton Promise is used.
+ */
+export class Mangata {
+  private static instance: Mangata
+  private apiPromise: Promise<ApiPromise> | null
+  private uri: string
+
+  /**
+   * The Mangata's constructor is private to prevent direct
+   * construction calls with the `new` operator.
+   */
+  private constructor(uri: string) {
+    this.apiPromise = null
+    this.uri = uri
+  }
+
+  /**
+   * Initialised via isReady & new with specific provider
+   */
+  private async connect() {
+    if (!this.apiPromise) {
+      const provider = new WsProvider(this.uri)
+      this.apiPromise = new ApiPromise(options({ provider })).isReady
+    }
+
+    return this.apiPromise
+  }
+
+  /**
+   * The static method that controls the access to the Mangata instance.
+   */
+  public static getInstance(uri: string): Mangata {
```

Comment:
I'm thinking of also providing a direct access to `api` object for the case if client for any reason wish to communicate with unwrapped api methods

Reply:
Yes definitely if we do not have method implemented they can do it yourself. I will add this method to class